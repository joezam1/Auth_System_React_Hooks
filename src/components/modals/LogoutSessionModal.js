import React, { useEffect, useState } from "react";
import SessionConfig from '../../../configuration/authentication/SessionConfig.js';
import LocalStorageService from "../../services/localStorage/LocalStorageService.js";

//Test: DONE
export default function LogoutSessionModal(){
    const [countdown, setCountdown] = useState();


    useEffect(()=>{
        console.log('LogoutSessionModal-useEffect-triggered-OK');
        let countdownValue = LocalStorageService.getItemFromLocalStorage(SessionConfig.LOGOUT_SESSION_COUNTDOWN_VALUE);
        setCountdown(countdownValue);
        let _intervalId = setInterval(() => {

            let countdownValueInsideInterval = LocalStorageService.getItemFromLocalStorage(SessionConfig.LOGOUT_SESSION_COUNTDOWN_VALUE);
            console.log('useEffect-setInterval-countdownValue', countdownValueInsideInterval);
            setCountdown(countdownValueInsideInterval);
            if(countdownValueInsideInterval <= 0){
                clearInterval(_intervalId);
                console.log('clearInterval-DONE-_intervalId', _intervalId);
                console.log('clearInterval-DONE-countdownValue', countdownValueInsideInterval);
            }
        }, SessionConfig.ONE_SECOND_IN_MILLISECONDS);
        console.log('LogoutSessionModal-useEffect-_intervalId', _intervalId)
    }, []);

    return(<div className="logout-session-modal-section">
        <div className="modal-container">
            <div className="modal-notification-box">
                <div className="modal-notification-text">
                    You will be redirected to Home in <b>{countdown ? countdown + ' seconds.': '' }</b>
                    <br/>
                </div>
            </div>
        </div>
    </div>);
}
