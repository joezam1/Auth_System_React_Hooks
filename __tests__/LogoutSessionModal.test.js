import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';

import LogoutSessionModal from "../src/components/modals/LogoutSessionModal.js";


describe('File: LogoutSessionModal.js', function(){
    afterEach(cleanup);
    afterAll(()=>{jest.resetAllMocks(); });

    describe('Component: LogoutSessionModal', function(){

        test('It can take a snapshot', function(){
            let container = render( <MemoryRouter> <LogoutSessionModal /> </MemoryRouter>);
            expect(container.asFragment(<LogoutSessionModal/> )).toMatchSnapshot();
        });
    });
});
