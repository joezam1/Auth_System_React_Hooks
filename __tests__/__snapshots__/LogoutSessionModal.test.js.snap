// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: LogoutSessionModal.js Component: LogoutSessionModal It can take a snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="logout-session-modal-section"
  >
    <div
      class="modal-container"
    >
      <div
        class="modal-notification-box"
      >
        <div
          class="modal-notification-text"
        >
          You will be redirected to Home in 
          <b />
          <br />
        </div>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
